File: 0997-find-the-town-judge/0997-find-the-town-judge.java
class Solution {
    public int findJudge(int N, int[][] trust) {
        int[] trustList = new int[N + 1];
        
        for (int[] person : trust) {
            trustList[person[1]]++;
            trustList[person[0]]--;
        }
        
        for(int i = 1; i <= N; i++) {
            if (trustList[i] == N - 1) {
                return i;
            }
        }
        return -1;
    }
}

File: 0997-find-the-town-judge/0997-find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        std::vector<int> pop(n+1);
        for(auto person : trust){
            pop[person[1]]++;
            pop[person[0]]--;
        }
        for(int i = 1; i <= n; i++) {
            if(pop[i] == n - 1) return i;
        }
        return -1;
    }
};